edit-post/assets/stylesheets/main.scss

```diff
@@ -219,18 +219,18 @@ body.gutenberg-editor-page {
 .editor-block-list__block {
 	input,
 	textarea {
-		// use opacity to work in various editor styles
+		// Use opacity to work in various editor styles.
 		&::-webkit-input-placeholder {
-			color: $dark-opacity-100;
+			color: $dark-opacity-300;
 		}
 		&::-moz-placeholder {
-			color: $dark-opacity-100;
+			color: $dark-opacity-300;
```

Comment:
Worth noting Firefox by default applies also some opacity on the placeholder. In Gutenberg the computed opacity is `1` just because it's inherited by [the WP rule in forms.css](https://core.trac.wordpress.org/browser/trunk/src/wp-admin/css/forms.css?rev=43309#L230). To make the Gutenberg styling really independent from core, there's the need to add `opacity: 1;` for any occurrence of the moz-placeholder.